dshr/dshr_mod.F90

```diff
@@ -258,57 +259,68 @@ subroutine dshr_mesh_init(gcomp, compid, logunit, compname, model_nxg, model_nyg
+    !     - if single column find the nearest neighbor in model_createmesh_fromfile
+    ! (2) if not single column - obtain the mesh directly from the mesh input
 
-    if (trim(model_meshfile) == 'undefined') then
+    if (trim(model_meshfile) == nullstr) then
 
-       ! TODO: verify that are only running on one processor here
-
-       call dshr_mesh_create_from_grid(trim(model_createmesh_fromfile), compid, model_mesh, rc=rc)
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-    else if (scmMode) then
-
-       ! verify that are only using 1 pe
-       if (my_task > 0) then
-          write(logunit,*) subname,' ERROR: scmmode must be run on one pe'
-          call shr_sys_abort(subname//' ERROR: scmmode2 tasks')
-       endif
-       ! obtain the single column lon and lat
-       call NUOPC_CompAttributeGet(gcomp, name='scmlon', value=cvalue, rc=rc)
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
-       read(cvalue,*) scmlon
-       call NUOPC_CompAttributeGet(gcomp, name='scmlat', value=cvalue, rc=rc)
+       ! Get single column values
+       call NUOPC_CompAttributeGet(gcomp, name='single_column', value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
-       read(cvalue,*) scmlat
-       if (my_task == master_task) then
-          write(logunit,*) ' scm mode, lon lat = ',scmmode, scmlon,scmlat
+       if (isPresent .and. isSet) then
+          read(cvalue,*) scol_mode
+          if (scol_mode) then
+             ! verify that ROF, WAV and LND are not trying to use single column mode
+             if (trim(compname) == 'ROF' .or. trim(compname) == 'LND' .or. trim(compname) == 'WAV') then
+                if (my_task == master_task) then
+                   write(logunit,*) subname,' ERROR: '//trim(compname)//' does not support single column mode '
+                end if
+                call shr_sys_abort(subname//' ERROR: '//trim(compname)//' does not support single column mode ')
+             end if
+             ! verify that are only using 1 pe
+             if (my_task > 0) then
+                if (my_task == master_task) then
+                   write(logunit,*) subname,' ERROR: single column mode must be run on one pe'
+                end if
+             endif
+             call shr_sys_abort(subname//' ERROR: single column mode must be run on 1 pe')
+
+             ! obtain the single column lon and lat
+             call NUOPC_CompAttributeGet(gcomp, name='scmlon', value=cvalue, rc=rc)
+             if (ChkErr(rc,__LINE__,u_FILE_u)) return
+             read(cvalue,*) scol_lon
+             call NUOPC_CompAttributeGet(gcomp, name='scmlat', value=cvalue, rc=rc)
+             if (ChkErr(rc,__LINE__,u_FILE_u)) return
+             read(cvalue,*) scol_lat
+             if (my_task == master_task) then
+                write(logunit,*) ' single column mode, lon lat = ',scol_mode, scol_lon, scol_lat
+             end if
+          else
+             scol_lon  = 1.e36_r8
```

Comment:
Use a defined constant here.  Should it be the netcdf double fill value?

Reply:
done